util/cluster.go

```diff
@@ -104,23 +117,22 @@ func CreateBackrestRepoSecrets(clientset *kubernetes.Clientset,
 		return err
 	}
 
-	// if an S3 key has been provided via the request, then use key and key secret inlcuded
-	// in the request instead of the default credentials provided by 'aws-s3-credentials.yaml'.
-	// If an S3 key doesn't exist in the request, the use `aws-s3-credentials.yaml`
-	var s3KeySecretData []byte
-	if backrestRepoConfig.BackrestS3Key != "" && backrestRepoConfig.BackrestS3KeySecret != "" {
-		s3Secret := AWSS3Secret{
-			backrestRepoConfig.BackrestS3Key,
-			backrestRepoConfig.BackrestS3KeySecret,
-		}
-		s3KeySecretData, err = yaml.Marshal(s3Secret)
-		if err != nil {
-			return err
-		}
-	} else {
-		s3KeySecretData = configs.Data["aws-s3-credentials.yaml"]
+	// if an S3 key has been provided via the request, then use key and key secret
+	// inlcuded in the request instead of the default credentials that are
+	// available in the Operator pgBackRest secret
+	backrestS3Key := []byte(backrestRepoConfig.BackrestS3Key)
```

Comment:
Just wondering if the syntax and logic might be a bit simpler/terse here if we stick with strings until adding the value to the map, e.g. sticking with string comparisons here:

```
if backrestRepoConfig.BackrestS3Key == "" {
    backrestS3Key = configs.Data[BackRestRepoSecretKeyAWSS3KeyAWSS3Key]
}

if backrestRepoConfig.backrestS3KeySecret == "" {
    backrestS3KeySecret = configs.Data[BackRestRepoSecretKeyAWSS3KeyAWSS3KeySecret]
}
```

and then just convert to a byte array when setting in the map, e.g.:

```
BackRestRepoSecretKeyAWSS3KeyAWSS3Key:       []byte(backrestS3Key)
BackRestRepoSecretKeyAWSS3KeyAWSS3KeySecret: []byte(backrestS3KeySecret),
```

Anyway, just a thought to remove a few lines and do a string comparison vs. using `len` on the byte array.  Ultimately the same result either way, so def no need to change this one.

Reply:
I settled on this, so it's already casted, so no need to cast further down.

```go
	// if an S3 key has been provided via the request, then use key and key secret
	// inlcuded in the request instead of the default credentials that are
	// available in the Operator pgBackRest secret
	backrestS3Key := []byte(backrestRepoConfig.BackrestS3Key)

	if backrestRepoConfig.BackrestS3Key == "" {
		backrestS3Key = configs.Data[BackRestRepoSecretKeyAWSS3KeyAWSS3Key]
	}

	backrestS3KeySecret := []byte(backrestRepoConfig.BackrestS3KeySecret)

	if backrestRepoConfig.BackrestS3KeySecret == "" {
		backrestS3KeySecret = configs.Data[BackRestRepoSecretKeyAWSS3KeyAWSS3KeySecret]
	}
```